### ISSUE TITLE
[clang-tidy] `move-const-arg` false positive

### ISSUE THREAD

**Issue by timblechmann** [llvm/llvm-project#61970]
I have lots of code where I'm moving objects into a lambda capture using the same name inside the capture as outside:

```cpp
auto foo(std::shared_ptr<int> i)
{
    return [i = std::move(i)]{};
}
```

Clang-Tidy gives me:
```console
warning: std::move of the const variable 'i' has no effect; remove std::move() or make the variable non-const [performance-move-const-arg]
```

I guess it refers to `i` as the variable in the lambda capture, not the variable (in the example the function parameter), which is moved into the capture

**Comment by carlosgalvezp** [comment]
What version of clang-tidy are you using? The problem doesn't show on latest trunk:
https://godbolt.org/z/MqMvehzW6

**Comment by timblechmann** [comment]
can't reproduce anymore after update
